File: python/problem1394-2.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        count = [0 for _ in range(501)]
        for a in arr:
            count[a] += 1
        for i in range(500, 0, -1):
            if count[i] == i:
                return i
        return -1

File: Java/problem1394-2.java
class Solution {
    public int findLucky(int[] arr) {
        int[] count = new int[501];
        for (int a : arr) {
            ++count[a];
        }
        for (int i = 500; i > 0; --i) {
            if (i == count[i]) {
                return i;
            }
        }
        return -1;
    }
}
